flang/lib/Semantics/resolve-names.cpp

```diff
@@ -8419,8 +8419,11 @@ void DeclarationVisitor::PointerInitialization(
     if (!context().HasError(ultimate)) {
       if (IsProcedurePointer(ultimate)) {
         auto &details{ultimate.get<ProcEntityDetails>()};
-        CHECK(!details.init());
-        if (const auto *targetName{std::get_if<parser::Name>(&target.u)}) {
+        if (details.init()) {
+          Say(name, "'%s' was previously initialized"_err_en_US);
```

Comment:
If there is an easy way, it would be nice to point at the previous initialization in an attachment (but If I am understanding details.init() correctly, you probably do not have that info here).

Reply:
I think the previous location is no longer available at this point; I couldn't figure out a way to find it, anyway.